catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/query/suggestion/MgrsCoordinateProcessor.java

```diff
@@ -0,0 +1,354 @@
+   *     MGRS was detected in the query, {@code null} is returned.
+   */
+  @Nullable
+  private LiteralSuggestion getMgrsSuggestion(String query) {
+    final List<String> matches = getMgrsCoordinateStrings(query);
+    if (matches.isEmpty()) {
+      LOGGER.debug("No valid MGRS strings were found or could be inferred from query [{}]", query);
+      return null;
+    }
+    final List<UsngCoordinate> uniqueMatches =
+        matches
+            .stream()
+            .map(this::parseMgrsString)
+            .filter(Objects::nonNull)
+            .collect(Collectors.toList());
+    final String name =
+        "MGRS: "
+            + uniqueMatches
+                .stream()
+                .map(mgrs -> "[ " + mgrs.toMgrsString() + " ]")
+                .collect(Collectors.joining(WHITESPACE));
+    if (uniqueMatches.size() == 1) {
+      UsngCoordinate mgrsCoord = uniqueMatches.get(0);
+      BoundingBox bb = translator.toBoundingBox(mgrsCoord);
+      return new LiteralSuggestion(LITERAL_SUGGESTION_ID, name, latLonFromBoundingBox(bb));
+    }
+    final List<LatLon> geo =
+        uniqueMatches
+            .stream()
+            .map(translator::toLatLon)
+            .map(d -> from(d.getLat(), d.getLon()))
+            .collect(Collectors.toList());
+    return new LiteralSuggestion(LITERAL_SUGGESTION_ID, name, geo);
+  }
+
+  @Nullable
+  private UsngCoordinate parseMgrsString(String mgrs) {
+    try {
+      return translator.parseMgrsString(mgrs);
+    } catch (ParseException e) {
+      LOGGER.debug(format("Detected string [%s] was not valid MGRS", mgrs), e);
+      return null;
+    }
+  }
+
+  private static List<LatLon> latLonFromBoundingBox(BoundingBox bb) {
+    return ImmutableList.of(
+        from(bb.getSouth(), bb.getWest()),
+        from(bb.getNorth(), bb.getWest()),
+        from(bb.getNorth(), bb.getEast()),
+        from(bb.getSouth(), bb.getEast()));
+  }
+
+  /**
+   * Given the user's query, return a list of valid MGRS strings that could be inferred from the
+   * query. The strings should be ready for submission to usng4j to be parsed and mathematically
+   * evaluated.
+   *
+   * <p>The query is terminated with whitespace so the user does not have to. See the LHS of {@link
+   * #PATTERN_MGRS_NUMERIC} for the cases where this is useful.
+   *
+   * @param query the input string potentially holding MGRS values.
+   * @return a list of valid MGRS values that can be sent to usng4j.
+   */
+  @VisibleForTesting
+  List<String> getMgrsCoordinateStrings(String query) {
+    final String terminatedQuery = query.concat(WHITESPACE);
+    final Matcher matcher = PATTERN_MGRS_ANYPART.matcher(terminatedQuery);
+    final List<String> mgrsStrings = new ArrayList<>();
+
+    StringBuilder mgrsBuilder = new StringBuilder();
+    MgrsPartType previousType = MgrsPartType.NONE;
+
+    while (matcher.find()) {
+      String mgrsPart = matcher.group().toUpperCase();
```

Comment:
Nope this is the correct commit. At the top of the file, the `PATTERN_MGRS_ANYPART` regex has been given the case insensitive flag. The others have **NOT** because during processing, if we encounter lowercase characters, we must convert them to uppercase so that the string is valid MGRS and suitable for submission to _usng4j_. Otherwise the library will choke.